### ISSUE TITLE
[8.4.0+] Not rendering columns and nestedHeaders when using updateSettings

### ISSUE THREAD

**Issue by gianadrias** [handsontable/handsontable#8587]
### Description
I used handsontable version 9.0.1 and it seems the updatesettings not working in columns and nestedheaders
but in version 8.3 it is working

### Steps to reproduce

1. On initial page load I set 2 columns
2. After init of handsontable I have a function that will push new column and nestedheader values then call updateSettings

### Demo

https://jsfiddle.net/2fhxne64/

### Your environment
* Handsontable version: 9.0.1
* Browser Name and version: Chrome

**Comment by swistach** [comment]
Hi @gianadrias
Thank you for your reporting. I confirm, since v8.4.0, updating settings using an empty dataset causes a problem with `nestedHeaders` (even if `dataSchema` is defined manually).

Comparison:
v8.3.2 (working): https://jsfiddle.net/4nb0zwxk/
v8.4.0+ (broken): https://jsfiddle.net/8tj5vnhr/
v9.0.2RC doesn't work either.

Column headers don't disappear if nestedHeaders is not in use. That means the problem occurs in the NestedHeaders plugin.

There is also another issue I reported here: #8589.

**Comment by zhaowenping001** [comment]
Hi, I also had this problem after upgrading my version recently, is there any way to avoid this problem?

**Comment by AMBudnik** [comment]
This regression hasn't been scheduled yet, @zhaowenping001 

However, I will notify you as soon as we fix this issue.

**Comment by budnix** [comment]
@zhaowenping001 I investigated the issue and it seems that it's not a regression. I think that the newest version of Handsontable and its _NestedHeaders_ plugin operates in a more strict mode. I mean that the dataset length (number of columns), has to be equal to or greater than the number of columns defined in the `nestedHeaders` setting.

I've updated your demo https://jsfiddle.net/budnix/dsaue6Lt/. I changed only 1 line. At 290 (the JavaScript tab) I've expanded the dataset with a new number of columns (previous value `var data = [{}];`, now `var data = Handsontable.helper.createSpreadsheetData(5, columns.columns.length);`).

Additionally, I've spotted that there is a problem with the `nestedRows: true` option. There is a problem with the _NestedRows_ plugin that throws an error after `updateSettings` method. I reported the issue in [a separate thread](https://github.com/handsontable/handsontable/issues/9018). In your demo, you're not using that plugin so for the sake of the simplicity of the demo, I allowed myself to turn this option off.
